pagopa/io-app
# chore: [IOBP-160] New design for credit card detail screen

## Short description
applied new design to credit card details screen, also refactored screens and components as they came to hand, with the aim to remove legacy things like class components and `connect` syntax 
<img width="250" alt="Screenshot 2023-08-01 at 15 43 56" src="https://github.com/pagopa/io-app/assets/60693085/b41da2e7-32ce-4ef1-95ef-d3640425a836">

## List of changes proposed in this pull request
-  took`IOLogoPaymentType` case mapping from idpay's scope and added it to IOLogoPayment in order to avoid casing problems that stem from formatting by different data sources
- added new `keysDecoderFromObject` util function, which takes an object and the `keyof typeof obj` type of it, and returns an io-ts decoder for a union of its keys as literals
- small `PaymentCardBig` refactor to best adress new problems that emerged in the component's first real-world usage
- updated its test snapsots
- updated `LogoPaymentOrDefaultIcon` to use the new map and a decoder generated by the new util func
- updated `IdpayInitiativesListComponent` to now handle the `isLast` case for the list
- huge refactor of `basePaymentMethodScreen`
- updated `PaymentMethodFeatures` to follow the design
- same for `PaymentMethodInitiatives`,  `PaymentMethodsSettings` and  `CreditCardDetailScreen` 


## How to test
tests should solve correctly, also navigate to a credit card's details screen and make sure everything works correctly.
also other kinds of payment methods should render effortlessly and with no visual glitches, although they dont completely follow design yet
